File: core/repository/repository_ingestion_engine.py
from __future__ import annotations

from pathlib import Path
from typing import Any, Dict, List

MAX_FILES = 100000

SUPPORTED_CODE_EXTENSIONS = {
    ".py",
    ".js",
    ".ts",
    ".tsx",
    ".jsx",
    ".go",
    ".rs",
    ".java",
    ".kt",
}

SKIP_DIR_NAMES = {
    ".git",
    "__pycache__",
    "node_modules",
    ".venv",
    "venv",
    "dist",
    "build",
}


def ingest_repository(path: str) -> Dict[str, Any]:
    root = Path(path)

    if not root.exists():
        return {
            "root": str(root),
            "files": [],
            "file_count": 0,
            "available": False,
            "reason": "path_not_found",
            "bounded": True,
        }

    files: List[Dict[str, Any]] = []

    count = 0

    for file in root.rglob("*"):
        if count >= MAX_FILES:
            break

        if not file.is_file():
            continue

        if any(part in SKIP_DIR_NAMES for part in file.parts):
            continue

        ext = file.suffix.lower()

        try:
            size = file.stat().st_size
        except OSError:
            size = 0

        files.append({
            "path": str(file),
            "extension": ext,
            "supported_code": ext in SUPPORTED_CODE_EXTENSIONS,
            "size": size,
        })

        count += 1

    return {
        "root": str(root),
        "files": sorted(files, key=lambda x: x["path"]),
        "file_count": len(files),
        "available": True,
        "bounded": True,
    }

File: src/repository/repositoryIngestionEngine.ts
/**
 * Converted from Python: core/repository/repository_ingestion_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_FILES: any = 100000;
export let SUPPORTED_CODE_EXTENSIONS: any = new Set([".py", ".js", ".ts", ".tsx", ".jsx", ".go", ".rs", ".java", ".kt"]);
export let SKIP_DIR_NAMES: any = new Set([".git", "__pycache__", "node_modules", ".venv", "venv", "dist", "build"]);
export function ingestRepository(path: any): any {
  var root: any = py.path(path);
  if (!py.truthy(root.exists())) {
    return {"root": py.toStr(root), "files": [], "file_count": 0, "available": false, "reason": "path_not_found", "bounded": true};
  }
  var files: any[] = [];
  var count: any = 0;
  var file: any;
  for (file of py.iter(root.rglob("*"))) {
    if (py.ge(count, MAX_FILES)) {
      break;
    }
    if (!py.truthy(file.is_file())) {
      continue;
    }
    if (py.any(py.iter(file.parts).map((part: any) => py.contains(SKIP_DIR_NAMES, part)))) {
      continue;
    }
    var ext: any = String(file.suffix).toLowerCase();
    try {
      var size: any = file.stat().st_size;
    } catch (_e: any) {
      size = 0;
    }
    py.listAppend(files, {"path": py.toStr(file), "extension": ext, "supported_code": py.contains(SUPPORTED_CODE_EXTENSIONS, ext), "size": size});
    count = py.add(count, 1);
  }
  return {"root": py.toStr(root), "files": py.sorted(files, {key: ((x: any) => py.at(x, "path")) as (item: any) => any}), "file_count": py.len(files), "available": true, "bounded": true};
}
